src/main/java/Edit_Group_Profile_Use_Case/EditGroupProfileRequestModel.java

```diff
@@ -0,0 +1,108 @@
+package Edit_Group_Profile_Use_Case;
+import java.util.HashMap;
+import java.util.*;
+
+public class EditGroupProfileRequestModel {
+    private String groupName;
+    private String courseCode;
+    private String description;
+    private String timeCommitment; //make sure not negative number
+    private String location;
+    private String meetingTime; //morning, afternoon, evening or day of the week
+
+    public EditGroupProfileRequestModel(String groupName, String description, String timeCommitment, String location, String meetingTime, String courseCode) {
+        this.groupName = groupName;
+        this.description = description;
+        this.timeCommitment = timeCommitment;
+        this.location = location;
+        this.meetingTime = meetingTime;
+        this.courseCode = courseCode;
+    }
+
+
+    public String getGroupName() {
+        return this.groupName;
+    }
+
+    public void setGroupName(String groupName) {
+        this.groupName = groupName;
+    }
+
+    public boolean setCourseCode(String courseCode) {
+        if (validateCourseCode(courseCode)) {
+            this.courseCode = courseCode;
+            return true;
+        }
+        return false;
+    }
+
+    public String getCourseCode() {
+        return this.courseCode;
+    }
+
```

Comment:
I think it would be better practice to include validateCourseCode() in the Interactor as the request model is just supposed to be a bundle of the input data. I think the responsibility to check validation is more in the interactor and that would follow Single Responsibility Principle better